src/FormErrorInterface.php

```diff
@@ -0,0 +1,102 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Form;
+
+/**
+ * FormErrors represents a form errors collection.
+ */
+interface FormErrorInterface
+{
+    /**
+     * Add error for the specified attribute.
+     *
+     * @param string $attribute attribute name.
+     * @param string $error attribute error message.
+     */
+    public function addError(string $attribute, string $error): void;
+
+    /**
+     * Removes errors for all attributes.
+     */
+    public function clear(): void;
+
+    /**
+     * Returns the errors for single attribute.
+     *
+     * @param string $attribute attribute name. Use null to retrieve errors for all attributes.
+     *
+     * @return array
+     */
+    public function getError(string $attribute): array;
+
+    /**
+     * Returns the errors for all attributes.
+     *
+     * @return array errors for all attributes or the specified attribute. null is returned if no error.
+     *
+     * Note that when returning errors for all attributes, the result is a two-dimensional array, like the following:
+     *
+     * ```php
+     * [
+     *     'username' => [
+     *         'Username is required.',
+     *         'Username must contain only word characters.',
+     *     ],
+     *     'email' => [
+     *         'Email address is invalid.',
+     *     ]
+     * ]
+     * ```
+     *
+     * {@see getFirstErrors()}
+     * {@see getFirstError()}
+     */
+    public function getErrors(): array;
+
+    /**
+     * Returns the errors for all attributes as a one-dimensional array.
+     *
+     * @param bool $showAllErrors boolean, if set to true every error message for each attribute will be shown otherwise
```

Comment:
How about introducing separate method `getFullErrorSummary()` instead of a boolean flag?

Reply:
done.